src/Compilers/CSharp/Portable/Symbols/FunctionPointers/FunctionPointerMethodSymbol.cs

```diff
@@ -111,48 +123,100 @@ static CallingConvention getCallingConvention(FunctionPointerCallingConventionSy
+                                if (!unmanagedList.ContainsDiagnostics)
+                                {
+                                    diagnostics.Add(ErrorCode.ERR_InvalidFunctionPointerCallingConvention, unmanagedList.OpenBracketToken.GetLocation(), "");
+                                }
+                                return CallingConvention.Default;
 
                             case { CallingConventions: var specifiers }:
-                                // PROTOTYPE(func-ptr): Handle multiple specifiers
-                                foreach (var convention in specifiers)
+                                // More than one identifier case
+                                checkUnmanagedSupport(typeBinder, callingConventionSyntax.ManagedOrUnmanagedKeyword.GetLocation(), diagnostics, isExtensionError: true);
+                                foreach (var specifier in specifiers)
                                 {
-                                    _ = reportBadConventionAndReturn(convention.Name.ValueText, convention.Name, diagnostics);
+                                    customModifiers.Add(handleIndividualUnrecognizedSpecifier(specifier, typeBinder, diagnostics));
                                 }
 
-                                return CallingConvention.Default;
+                                return CallingConvention.Unmanaged;
                         }
 
                     case var unexpected:
                         throw ExceptionUtilities.UnexpectedValue(unexpected);
                 }
 
-                static CallingConvention reportBadConventionAndReturn(string givenConvention, SyntaxNodeOrToken node, DiagnosticBag diagnostics)
+                static CallingConvention handleSingleConvention(FunctionPointerUnmanagedCallingConventionSyntax specifier, Binder typeBinder, ArrayBuilder<CustomModifier> customModifiers, DiagnosticBag diagnostics)
+                {
+                    checkUnmanagedSupport(typeBinder, specifier.GetLocation(), diagnostics, isExtensionError: true);
+                    customModifiers.Add(handleIndividualUnrecognizedSpecifier(specifier, typeBinder, diagnostics));
+                    return CallingConvention.Unmanaged;
+                }
+
+                static CustomModifier handleIndividualUnrecognizedSpecifier(FunctionPointerUnmanagedCallingConventionSyntax specifier, Binder typeBinder, DiagnosticBag diagnostics)
+                {
+                    const string CallingConventionMetadataFormatString = "System.Runtime.CompilerServices.CallConv{0}";
+                    string specifierText = specifier.Name.ValueText;
+                    var metadataName = string.Format(CultureInfo.InvariantCulture, CallingConventionMetadataFormatString, specifierText);
+
+                    var specifierType = typeBinder.Compilation.Assembly.CorLibrary.GetTypeByMetadataName(metadataName, includeReferences: false, isWellKnownType: true, conflicts: out _, warnings: null);
```

Comment:
>true [](start = 162, length = 4)

This is not a well known type

Reply:
Yes, but I believe we want it to behave like a well-known type for all intents and purposes. This flag checks that the symbol is accessible and public.

---
In reply to: [457723185](https://github.com/dotnet/roslyn/pull/46110#discussion_r457723185) [](ancestors = 457723185)

Reply:
> Yes, but I believe we want it to behave like a well-known type for all intents and purposes. This flag checks that the symbol is accessible and public.

We do want them to be public and we can check for that. We do not want them to be treated as well-known even if at the moment the requirements appear to match. Also, in my other comment I mentioned that I think we should use a different lookup API, that provides guarantees that we would want. 

---
In reply to: [457785546](https://github.com/dotnet/roslyn/pull/46110#discussion_r457785546) [](ancestors = 457785546,457723185)

Reply:
> Also, in my other comment I mentioned that I think we should use a different lookup API, that provides guarantees that we would want. 

I don't believe it does: this codepath uses the API you called out, and then explicitly checks the return type to ensure it's public and visible.

---
In reply to: [458310033](https://github.com/dotnet/roslyn/pull/46110#discussion_r458310033) [](ancestors = 458310033,457785546,457723185)

Reply:
> I don't believe it does: this codepath uses the API you called out, and then explicitly checks the return type to ensure it's public and visible.

The guarantees that I refer to are mentioned in that other comment, they are not mentioned on this thread. If you are referring to the public accessibility check, I think we want to perform this check here explicitly and report good error that the type such-and-such is not public instead of trying to figure out why ```GetTypeByMetadataName``` failed to locate the type .

---
In reply to: [458369877](https://github.com/dotnet/roslyn/pull/46110#discussion_r458369877) [](ancestors = 458369877,458310033,457785546,457723185)